### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by FatCaterpillar**

### CODE REVIEW

File: refined-types/src/test/kotlin/com/virtuslab/basetypes/refined/DigitTest.kt [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+import io.kotlintest.properties.Gen
+import io.kotlintest.properties.forAll
+import io.kotlintest.specs.StringSpec
+
+internal class DigitTest : StringSpec() {
+    init {
+        "of(Int)" {
+            forAll(Gen.positiveIntegers()) { a: Int ->
+                Digit.of(a).nonEmpty()
```
Comment (krzykrucz) [inline refined-types/src/test/kotlin/com/virtuslab/basetypes/refined/DigitTest.kt]: I don't understand this test, what is it supposed to do? specifically Gen.positiveintegers

at first glance it looks like it is testing if a valid digit can be created from all positive integers

File: result-arrow/src/main/kotlin/com/virtuslab/basetypes/result/arrow/AsyncResult.kt [hunk 2]
```diff
@@ -12,29 +12,29 @@ fun <T : Any, E : Exception> Result<T, E>.liftAsync(): AsyncResult<T, E> =
...
+infix fun <T1 : Any, T2 : Any, E : Exception> AsyncResult<T1, E>.then(f: (T1) -> AsyncResult<T2, E>): AsyncResult<T2, E> =
     this.flatMap { result1 ->
         when (result1) {
             is Success -> f(result1.value)
             is Failure -> Failure(result1.error).liftAsync()
         }
     }
 
-infix fun <T1 : Any, T2 : Any, E : Exception> Result<T1, E>.then(f: (T1) -> AsyncResult<T2, E>): AsyncResult<T2, E> =
+infix fun <T1 : Any, T2 : Any, E : Exception> Result<T1, E>.thenDoAsync(f: (T1) -> AsyncResult<T2, E>): AsyncResult<T2, E> =
```
Comment (krzykrucz) [inline result-arrow/src/main/kotlin/com/virtuslab/basetypes/result/arrow/AsyncResult.kt]: maybe thenAsync? as in line 21

File: refined-types/src/test/kotlin/com/virtuslab/basetypes/refined/DigitTest.kt (hunk 1 above)
Comment (krzykrucz) [reply refined-types/src/test/kotlin/com/virtuslab/basetypes/refined/DigitTest.kt]: Ah shit, the test is ok. The Digit should be less or equal 9, though. My mistake

Comment (krzykrucz) [reply refined-types/src/test/kotlin/com/virtuslab/basetypes/refined/DigitTest.kt]: Sorry, I could not resist fixing this :D Please git pull

File: result-arrow/src/main/kotlin/com/virtuslab/basetypes/result/arrow/AsyncResult.kt (hunk 2 above)
Comment (FatCaterpillar) [reply result-arrow/src/main/kotlin/com/virtuslab/basetypes/result/arrow/AsyncResult.kt]: Done

File: refined-types/src/test/kotlin/com/virtuslab/basetypes/refined/DigitTest.kt (hunk 1 above)
Comment (FatCaterpillar) [reply refined-types/src/test/kotlin/com/virtuslab/basetypes/refined/DigitTest.kt]: Cool :D

**Pull request merged**